test/test-rotatesTetrominos.js

```diff
@@ -1,32 +1,194 @@
-            
+
+        determineShift (typeOfTetromino, rotationPhaseOfTetromino) {
+            if (typeOfTetromino === "O") {
+                if (rotationPhaseOfTetromino === "stable") {
+                    return { x: 0, y: 1 };
+                }
+            }
+            if (typeOfTetromino === "J") {
+                if (rotationPhaseOfTetromino === "up") {
+                    return { x: 0, y: 1 };
+                }
+                if (rotationPhaseOfTetromino === "left") {
+                    return { x: 0, y: 2 };
+                }
+                if (rotationPhaseOfTetromino === "down") {
+                    return { x: 0, y: 1 };
+                }
+                if (rotationPhaseOfTetromino === "right") {
+                    return { x: 0, y: 2 };
+                }
+            }
+            if (typeOfTetromino === "L") {
+                if (rotationPhaseOfTetromino === "up") {
+                    return { x: 0, y: 1 };
+                }
+                if (rotationPhaseOfTetromino === "left") {
+                    return { x: 0, y: 2 };
+                }
+                if (rotationPhaseOfTetromino === "down") {
+                    return { x: 0, y: 1 };
+                }
+                if (rotationPhaseOfTetromino === "right") {
+                    return { x: 0, y: 2 };
+                }
+            }
+            if (typeOfTetromino === "T") {
+                if (rotationPhaseOfTetromino === "up") {
+                    return { x: 0, y: 2 };
+                }
+                if (rotationPhaseOfTetromino === "left") {
+                    return { x: 0, y: 2 };
+                }
+                if (rotationPhaseOfTetromino === "down") {
+                    return { x: 1, y: 1 };
+                }
+                if (rotationPhaseOfTetromino === "right") {
+                    return { x: -1, y: 1 };
+                }
+            }
+            if (typeOfTetromino === "S") {
+                if (rotationPhaseOfTetromino === "horizontal") {
+                    return { x: 0, y: 2 };
+                }
+                if (rotationPhaseOfTetromino === "vertical") {
+                    return { x: 0, y: 1 };
+                }
+            }
+            if (typeOfTetromino === "Z") {
+                if (rotationPhaseOfTetromino === "horizontal") {
+                    return { x: 0, y: 2 };
+                }
+                if (rotationPhaseOfTetromino === "vertical") {
+                    return { x: 0, y: 1 };
+                }
+            }
+            if (typeOfTetromino === "I") {
+                if (rotationPhaseOfTetromino === "horizontal") {
+                    return { x: 1, y: 2 };
+                }
+                if (rotationPhaseOfTetromino === "vertical") {
+                    return { x: -1, y: 1 };
+                }
+            }
+        } 
     }
     it("L-tetromino rotation", function(){
         let typeTetromino = "L";
+        let phase = "rihgt";
```

Comment:
typo

Reply:
Where is the typo?
All words is correct